cni/pkg/nodeagent/server_linux.go

```diff
@@ -53,34 +58,67 @@ func initMeshDataplane(client kube.Client, args AmbientArgs) (*meshDataplane, er
 		return nil, fmt.Errorf("error initializing the ztunnel server: %w", err)
 	}
 
-	hostIptables, podIptables, err := iptables.NewIptablesConfigurator(
-		hostCfg,
-		podCfg,
-		realDependenciesHost(),
-		realDependenciesInpod(UseScopedIptablesLegacyLocking),
-		iptables.RealNlDeps(),
-	)
+	hostTrafficManager, podTrafficManager, err := trafficmanager.NewTrafficRuleManager(&trafficmanager.TrafficRuleManagerConfig{
+		NativeNftables: args.NativeNftables,
+		HostConfig:     hostCfg,
+		PodConfig:      podCfg,
+		HostDeps:       realDependenciesHost(),
+		PodDeps:        realDependenciesInpod(UseScopedIptablesLegacyLocking),
+		NlDeps:         iptables.RealNlDeps(),
+	})
 	if err != nil {
-		return nil, fmt.Errorf("error configuring iptables: %w", err)
+		return nil, fmt.Errorf("error creating traffic managers: %w", err)
 	}
 
 	// Create hostprobe rules now, in the host netns
-	hostIptables.DeleteHostRules()
+	hostTrafficManager.DeleteHostRules()
```

Comment:
I think this is not needed if nftables is enabled since it will automatically flush pre-existing objects

Reply:
Right, but trafficmanager is a higher-level abstraction over `iptables` and `nftables`. Here, we’re invoking the hostTrafficManager APIs to delete the host rules, which is required for `iptables` and somewhat optional for `nftables`.

Reply:
I see your point, let me update it.